perses/annihilation/new_relative.py

```diff
@@ -0,0 +1,1213 @@
+        return electrostatics_addition
+
+    def _nonbonded_custom_nocutoff(self):
+        """
+        Get a part of the nonbonded energy expression when there is no cutoff.
+
+        Returns
+        -------
+        sterics_energy_expression : str
+            The energy expression for U_sterics
+        electrostatics_energy_expression : str
+            The energy expression for electrostatics
+        """
+        # soft-core Lennard-Jones
+        sterics_energy_expression = "U_sterics = 4*epsilon*x*(x-1.0); x = (sigma/reff_sterics)^6;"
+        # soft-core Coulomb
+        electrostatics_energy_expression = "U_electrostatics = ONE_4PI_EPS0*chargeprod/reff_electrostatics;"
+        return sterics_energy_expression, electrostatics_energy_expression
+
+    def _nonbonded_custom_cutoff(self, epsilon_solvent, r_cutoff):
+        """
+        Get the energy expressions for sterics and electrostatics under a reaction field assumption.
+
+        Parameters
+        ----------
+        epsilon_solvent : float
+            The reaction field dielectric
+        r_cutoff : float
+            The cutoff distance
+
+        Returns
+        -------
+        sterics_energy_expression : str
+            The energy expression for U_sterics
+        electrostatics_energy_expression : str
+            The energy expression for electrostatics
+        """
+        # soft-core Lennard-Jones
+        sterics_energy_expression = "U_sterics = 4*epsilon*x*(x-1.0); x = (sigma/reff_sterics)^6;"
+
+        electrostatics_energy_expression = "U_electrostatics = ONE_4PI_EPS0*chargeprod*(reff_electrostatics^(-1) + k_rf*reff_electrostatics^2 - c_rf);"
+        k_rf = r_cutoff**(-3) * ((epsilon_solvent - 1) / (2*epsilon_solvent + 1))
+        c_rf = r_cutoff**(-1) * ((3*epsilon_solvent) / (2*epsilon_solvent + 1))
+        electrostatics_energy_expression += "k_rf = %f;" % (k_rf / k_rf.in_unit_system(unit.md_unit_system).unit)
+        electrostatics_energy_expression += "c_rf = %f;" % (c_rf / c_rf.in_unit_system(unit.md_unit_system).unit)
+        return sterics_energy_expression, electrostatics_energy_expression
+
+    def _nonbonded_custom_ewald(self, alpha_ewald, delta, r_cutoff):
+        """
+        Get the energy expression for Ewald treatment.
+
+        Parameters
+        ----------
+        alpha_ewald : float
+            The Ewald alpha parameter
+        delta : float
+            The PME error tolerance
+        r_cutoff : float
+            The cutoff distance
+
+        Returns
+        -------
+        sterics_energy_expression : str
+            The energy expression for U_sterics
+        electrostatics_energy_expression : str
+            The energy expression for electrostatics
+        """
+        # soft-core Lennard-Jones
+        sterics_energy_expression = "U_sterics = 4*epsilon*x*(x-1.0); x = (sigma/reff_sterics)^6;"
+        if alpha_ewald == 0.0:
```

Comment:
`alpha_ewald` here may be a `Quantity` object, in which case could always evaluates to `False` even if its value is 0.0. Probably worth checking.